homeassistant/components/group/__init__.py

```diff
@@ -323,6 +342,7 @@ async def groups_service_handler(service):
                     vol.Optional(ATTR_ALL): cv.boolean,
                     vol.Exclusive(ATTR_ENTITIES, "entities"): cv.entity_ids,
                     vol.Exclusive(ATTR_ADD_ENTITIES, "entities"): cv.entity_ids,
+                    vol.Exclusive(ATTR_REMOVE_ENTITIES, "entities"): cv.entity_ids,
```

Comment:
This can't be `exclusive`, because that means you can't both add and remove in the same call, while that is something that is supported by the code.

Reply:
What would you suggest as an alternative here? I'm not very familiar with Voluptuous schemas. Is there a way to express that add and remove are jointly exclusive with everything else? Other options are (1) remove the exclusivity behavior for `add` as well (not ideal), or (2) go back to only allowing `add` or `remove`, not both in one service call.

Reply:
You can't express that in the dict schema, but you can add another validator.

```python
def validate_entities(val):
    if ATTR_ENTITIES in val and (ATTR_ADD_ENTITIES in val or ATTR_REMOVE_ENTITIES in val):
        raise vol.Invalid(f'Cannot combine "{ATTR_ENTITIES}" with add/remove')

vol.All(
  vol.Schema({…}),
  validate_entities,
)
```